From 86cb2f81adcfb1b4ae490c7dffdaa7c83c3687de Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:50 -0500
Subject: [drm] radeon: fix active_cu mask on SI and CIK after re-init (v3)

Message-id: <1418852380-13061-40-git-send-email-rclark@redhat.com>
Patchwork-id: 102217
O-Subject: [RHEL7 drm 39/89] drm/radeon: fix active_cu mask on SI and CIK after re-init (v3)
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.16.4

commit 907ed7ee43a3fee9759f99f163fe34754e49a006

Author:     Alex Deucher <alexander.deucher@amd.com>
AuthorDate: Tue Aug 19 11:56:38 2014 -0400
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:19:49 2014 -0500

    drm/radeon: fix active_cu mask on SI and CIK after re-init (v3)

    commit 52da51f0f9ea9d213adfc99223630707b26d1d38 upstream.

    Need to initialize the mask to 0 on init, otherwise it
    keeps increasing.

    bug:
    https://bugzilla.kernel.org/show_bug.cgi?id=82581

    v2: also fix cu count
    v3: split count fix into separate patch

    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
    Reviewed-by: Michel Danzer <michel.daenzer@amd.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/radeon/cik.c b/drivers/gpu/drm/radeon/cik.c
index a5ce5e6..78b4a27 100644
--- a/drivers/gpu/drm/radeon/cik.c
+++ b/drivers/gpu/drm/radeon/cik.c
@@ -3449,6 +3449,7 @@ static void cik_gpu_init(struct radeon_device *rdev)
        rdev->config.cik.max_sh_per_se,
        rdev->config.cik.max_backends_per_se);
 
+ rdev->config.cik.active_cus = 0;
  for (i = 0; i < rdev->config.cik.max_shader_engines; i++) {
   for (j = 0; j < rdev->config.cik.max_sh_per_se; j++) {
    rdev->config.cik.active_cus +=
diff --git a/drivers/gpu/drm/radeon/si.c b/drivers/gpu/drm/radeon/si.c
index 238ff93..6c17d3b 100644
--- a/drivers/gpu/drm/radeon/si.c
+++ b/drivers/gpu/drm/radeon/si.c
@@ -3099,6 +3099,7 @@ static void si_gpu_init(struct radeon_device *rdev)
        rdev->config.si.max_sh_per_se,
        rdev->config.si.max_cu_per_sh);
 
+ rdev->config.si.active_cus = 0;
  for (i = 0; i < rdev->config.si.max_shader_engines; i++) {
   for (j = 0; j < rdev->config.si.max_sh_per_se; j++) {
    rdev->config.si.active_cus +=
-- 
1.7.1